### PR TITLE
[AIRFLOW-272] add DagRunStateSensor

### PR DESCRIPTION
**Pull request by NarekA**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- [AIRFLOW-272](https://issues.apache.org/jira/browse/AIRFLOW-272)

Testing Done:
Added 2 unit-tests for this.

### CODE REVIEW

**Comment by NarekA** [comment]
This being my first contribution to this project, let me know if I'm doing anything wrong.

File: airflow/operators/sensors.py [hunk 1]
```diff
@@ -222,6 +222,56 @@ def poke(self, context):
...
+            allowed_states=None,
+            execution_delta=None,
+            *args, **kwargs):
+        super(DagRunStateSensor, self).__init__(*args, **kwargs)
+        self.allowed_states = allowed_states or [State.SUCCESS]
+        self.execution_delta = execution_delta
+        self.external_dag_id = external_dag_id
+
+    def poke(self, context):
+        if self.execution_delta:
```
Comment (gwax) [inline airflow/operators/sensors.py:252]: probably best to be explicit:

``` python
if self.execution_delta is not None:
    ...
```

**Comment by ldct** [comment]
Hi @NarekA, thanks for the PR. Can you describe the use case for this? Also, can you rebase the PR?

**Comment by ldct** [comment]
@NarekA as part of the cold case PR cleanup effort, please respond within 5 days or we will close this PR for inactivity

**Pull request closed without merging**